### PR TITLE
Issue-15: Add support for Maybe#filter(Predicate)

### LINKED ISSUE #15: Add support for Maybe#filter(Predicate)
* Passing the `predicate` returns the `Just` of the value.
* Failing the `predicate` or being a `Nothing` returns `Nothing`.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #15
SemVer:Minor
enhancement

### CODE REVIEW

File: docs/data/Maybe/Maybe.md [hunk 1]
```diff
@@ -25,6 +25,9 @@
 
 ## Instance Methods
 
+Instance methods that accept and argument have curried implementations provided by the `Maybes` class where the instance is always the last argument.
```
Comment (doomchild) [inline docs/data/Maybe/Maybe.md]: `accept and argument` → `accept an argument`

**Review by doomchild — CHANGES_REQUESTED** [review]
Grammatical change.

File: docs/data/Maybe/Maybe.md (hunk 1 above)
Comment (jlmorgan) [reply docs/data/Maybe/Maybe.md]: Fixed.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Pull request merged**